packages/react-instantsearch-hooks-dom/src/ui/Highlight.tsx

```diff
@@ -37,24 +39,43 @@ type HighlightedPart = {
   value: string;
 };
 
+export type HighlightClassNames = {
+  /**
+   * Class names to apply to the root element
+   */
+  root: CSSClass;
+  /**
+   * Class names to apply to the highlighted parts
+   */
+  highlighted: CSSClass;
+  /**
+   * Class names to apply to the non-highlighted parts
+   */
+  nonHighlighted: CSSClass;
+  /**
+   * Class names to apply to the separator between highlighted parts
+   */
+  separator: CSSClass;
+};
+
 export type HighlightProps = React.HTMLAttributes<HTMLSpanElement> & {
-  baseClassName: string;
+  classNames: HighlightClassNames;
   highlightedTagName?: React.ReactType;
   nonHighlightedTagName?: React.ReactType;
   separator?: React.ReactNode;
   parts: HighlightedPart[][];
 };
 
 export function Highlight({
-  baseClassName,
   parts,
   highlightedTagName = 'mark',
   nonHighlightedTagName = 'span',
   separator = ', ',
+  classNames,
```

Comment:
This component doesn't support the original `className` prop, as opposed to all the other components. I think we should remove `classNames.root` and replace it with the `className` prop then. Wdyt?

Reply:
see also https://github.com/algolia/react-instantsearch/pull/3379#discussion_r825977066